sensors/lego_sensor_class.c

```diff
@@ -511,6 +517,17 @@ static ssize_t fw_version_show(struct device *dev, struct device_attribute *attr
 	return snprintf(buf, LEGO_SENSOR_FW_VERSION_SIZE + 2, "%s\n", sensor->fw_version);
 }
 
+static ssize_t text_value_show(struct device *dev, struct device_attribute *attr,
+			      char *buf)
+{
+	struct lego_sensor_device *sensor = to_lego_sensor_device(dev);
+
+	if (!sensor->text_value_read)
+		return -EOPNOTSUPP;
+
+        return sensor->text_value_read(buf, PAGE_SIZE);
```

Comment:
All implementations will have to copy the value to `buf`, so better to do it here. `text_value_read` should return `const char *` or `PTR_ERR` (or is it `ERR_PTR`?). Then do `spintf(buf, "%s\n", ret)` here if no error.